python/erdos/streams.py

```diff
@@ -58,6 +59,147 @@ def name(self) -> str:
     def name(self, name: str):
         self._internal_stream.set_name(name)
 
+    def map(self, function: Callable[[Any], Any]) -> "OperatorStream":
+        """Applies the given function to each received value on the stream, and outputs
+        the results on the returned stream.
+
+        Args:
+            function (Callable[[Any], Any]): The function to be applied on each message
+                received on the input stream.
+
+        Returns:
+            An :py:class:`OperatorStream` that carries the results of the applied
+            function.
+        """
+
+        def map_fn(serialized_data: bytes) -> bytes:
+            result = function(pickle.loads(serialized_data))
+            return pickle.dumps(result)
+
+        return OperatorStream(self._internal_stream._map(map_fn))
+
+    def flat_map(self, function: Callable[[Any], Sequence[Any]]) -> "OperatorStream":
```

Comment:
Could you add a todo to optimize the Python flat map implementation using iterators? In this implementation, all new elements are generated and then sent; whereas the Rust implementation sends messages eagerly after each element is generated.